democracy/views/utils.py

```diff
@@ -97,3 +99,56 @@ def filter_by_hearing_visible(queryset, request, hearing_lookup='hearing'):
         filters['%s__published' % hearing_lookup] = True
 
     return queryset.filter(**filters)
+
+
+class JSONPKRelatedField(PrimaryKeyRelatedField):
+    """
+    Support of showing and saving of expanded JSON nesting or just a resource
+    ID.
+    Serializing is controlled by query string param 'expand', deserialization
+    by format of JSON given.
+    Default serializing is expand=false.
+    """
+
+    invalid_json_error = _('Incorrect JSON. Expected JSON, received %s.')
+    missing_id_error = _('The primary key is missing. Expected {"id": id, ...}, received %s.')
+
+    def __init__(self, *args, **kwargs):
+        self.related_serializer = kwargs.pop('serializer', None)
+        self.expanded = kwargs.pop('expanded', False)
+        super(JSONPKRelatedField, self).__init__(*args, **kwargs)
+
+    def use_pk_only_optimization(self):
+        if self.is_expanded():
+            return False
+        else:
+            return True
+
+    def to_representation(self, obj):
+        if isinstance(self.related_serializer, str):
+            self.related_serializer = globals().get(self.related_serializer, None)
+        if self.is_expanded():
+            return self.related_serializer(obj, context=self.context).data
+        id_ = super(JSONPKRelatedField, self).to_representation(obj)
+        if id_ is None:
+            return None
+        return {
+            'id': id_,
+        }
+
+    def to_internal_value(self, value):
+        if not isinstance(value, dict):
+            raise ValidationError(self.invalid_json_error % type(value).__name__)
+        if 'id' not in value:
+            raise ValidationError(self.missing_id_error % value)
+
+        id_ = value['id']
+        if not id_:
+            if self.required:
+                raise ValidationError(_('This field is required.'))
+            return None
+
+        return super().to_internal_value(id_)
+
+    def is_expanded(self):
```

Comment:
Is this necessary? `expanded` will always be an attribute of a JSONPKRelatedField instance.